File: c_cpp/quicksort.cpp
#include<iostream>
int Partition(int *arr, int low, int high)
{
    int tmp = arr[low];
    while (low < high)
    {
        while (low < high && arr[high] > tmp)
        {
            high--;
        }
        if (low < high)
        {
            arr[low] = arr[high];
        }
        while (low < high && arr[low] <= tmp)
        {
            low++;
        }
        if (low < high)
        {
            arr[high] = arr[low];
        }
    }
    arr[low] = tmp;
    return low;
}
void Quick(int *arr, int low, int high)
{
    int par = Partition(arr, low, high);
    if (low < par - 1) 
    {
        Quick(arr, low, par - 1);
    }
    if (par + 1 < high) 
    {
        Quick(arr, par + 1, high);
    }
}
void QuickSort(int *arr, int len)
{
    Quick(arr, 0, len - 1); 
}

int main(){
    std::cout<<"Please input 5 numbers:"<<std::endl;
    int a[5];
    for (int i = 0; i < 5; i++)
    {
        scanf("%d",&a[i]);
    }
    QuickSort(a,5);
    for (int i = 0; i < 5; i++)
    {
        std::cout<<a[i]   ;
    }
}

File: go/quicksort.go
package main

import "fmt"

func QuickSort(array []int) {
	Quick(array, 0, len(array)-1)
}
 
func Quick(array []int, l, r int) {
	if l >= r {
		return
	}
	point := Partition(array, l, r)
	Quick(array, l, point-1)
	Quick(array, point+1, r)
}
 

func Partition(array []int, l, r int) int {
	point := array[r]
	i := l
	for j := l; j < r; j++ {
		if array[j] < point {
			array[i], array[j] = array[j], array[i]
			i++
		}
	}
	array[i], array[r] = array[r], array[i]
	return i
}

func main(){
	fmt.Println("Please input 5 numbers:");
	var a  [] int
	var n int
	for i := 0; i < 5; i++{
		fmt.Scan(&n)
		a = append(a, n)
	}	
	QuickSort(a)
	for _, v := range a {
		fmt.Printf("%5d",  v)
	}
}